### PR TITLE
Add some new g2-styled icons to the Icons package. #20284

### PR DESCRIPTION
**Pull request by jameskoster**
## Description
Proposing a variety of new icons based on the G2 design language, plus an update to the `page` icon.

## How has this been tested?
Testing is required

## Screenshots
icons

The icons I'm proposing are in the right-hand column. [Figma link](https://www.figma.com/file/fnyj380i05vGzuuH60frLQ/G2-Design?node-id=3234%3A24162).

## Types of changes
New feature

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by pablohoneyhoney** [comment]
Nice!

Here some notes:

- Percent, payment, trending, product looking solid. 
- Shipping, Money, Receipt could use further reduction. 
- Personalize and Institution don't think are working well. Are there other market conventions you've explored? 
- Widgets seems the same as _Categories_ currently. 
- Review could be filled and smaller. 
- Page and Inbox could have a bit more breadth between lines.

**Comment by jasmussen** [comment]
Thanks for the PR! 

It's a little hard to tell from the screenshot — did you adjust the stroke-widths of the adjusted icons? While it isn't a hard rule for any of the new icons, the 1.5px stroke most of them come with ties in very well with the dark outlines of the new block UI. If you can point me to a Figma file, I can help make this thinner stroke width happen.

**Comment by jameskoster** [comment]
> Personalize and Institution don't think are working well. Are there other market conventions you've explored?

Lots of options we can explore for personalize. Paint brush / tin, etc, or something more abstract. 

Institute is tricky. It's inspired by the `account_balance` icon in Material (which we've been utilising as a bank in our UI), and the [majority of other iconographic representations of this term](https://www.iconfinder.com/search/?q=institute&from=homepage). There are probably some variations I can come up with though.

> Widgets seems the same as Categories currently.

Hah, I didn't spot that one. Will pull the stock proposal.

> Review could be filled and smaller.
> Page and Inbox could have a bit more breadth between lines.

Will adjust this week.

> It's a little hard to tell from the screenshot — did you adjust the stroke-widths of the adjusted icons? While it isn't a hard rule for any of the new icons, the 1.5px stroke most of them come with ties in very well with the dark outlines of the new block UI. If you can point me to a Figma file, I can help make this thinner stroke width happen.

All the strokes are 1.5px. There's a Figma link in the op :)

**Comment by jasmussen** [comment]
> All the strokes are 1.5px. There's a Figma link in the op :)

I don't know how I missed that! 🤦‍♂ 

Thanks!

**Comment by jameskoster** [comment]
Here are some revisions / options based on the feedback above. Check the "G2 i2" column:

icons

* Money is tough. Without the flourishes on the corners I worry the bills look more like keyholes or something. I didn't want to use any currency symbols to avoid potential localisation issues, but if we want to be super clear / simple, that might be something we have to explore. I wonder if we can localise an icon? :D
* I made the review icon smaller, and filled it. Personally I prefer this outlined rather than filled - I think it is more consistent that way. Happy to go with the consensus though.
* Institute is tough. Added some options there, but to be honest I prefer the original.

[Figma link](https://www.figma.com/file/fnyj380i05vGzuuH60frLQ/G2-Design?node-id=3234%3A24162) again if anyone would like to riff on these.

**Comment by pablohoneyhoney** [comment]
I like the revisions @jameskoster 

**Money**: would it be simpler as a coin and the dollar sign inside?
**Institution**: agree let's stick with the original then.
**Review**: looking good but seems a bit too rounded, could it be less so?
**Receipt**: I prefer this one. 
Screen Shot 2020-03-30 at 1 06 35 PM

**Personalize**: I wonder if the brush I created in the file could work. 
block-brush

**Comment by jameskoster** [comment]
Another update:

icons

I think if we call the money icon "Dollar" instead then we can use the currency symbol. That leaves scope to add more currencies in the future.

Made the star slightly less rounded.

> Personalize: I wonder if the brush I created in the file could work.

Looks a little like a mop to my eye, but it could work.

**Comment by pablohoneyhoney** [comment]
Nice, we can retouch the brush for sure. 
The currencies look good, maybe smaller within the circle.

**Comment by jameskoster** [comment]
Good call, I shrunk the symbols down a bit. Here's the final set, hopefully we're good to go now :)

icons

File: packages/icons/src/library/page.js [hunk 1]
```diff
@@ -4,8 +4,8 @@
 import { SVG, Path } from '@wordpress/primitives';
 
 const page = (
-	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="-2 -2 24 24">
-		<Path d="M6 15V2h10v13H6zm-1 1h8v2H3V5h2v11z" />
+	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+		<path d="M7 5.5h10a.5.5 0 01.5.5v12a.5.5 0 01-.5.5H7a.5.5 0 01-.5-.5V6a.5.5 0 01.5-.5zM17 4H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V6a2 2 0 00-2-2zm-1 3.75H8v1.5h8v-1.5zM8 11h8v1.5H8V11zm6 3.25H8v1.5h6v-1.5z" />
```
Comment (jasmussen) [inline packages/icons/src/library/page.js:8]: Capital P (dangit 😉)

File: packages/icons/src/library/inbox.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+ * WordPress dependencies
+ */
+import { SVG, Path } from '@wordpress/primitives';
+
+const inbox = (
+	<SVG xmlns="http://www.w3.org/2000/SVG" viewBox="0 0 24 24">
+		<path fill-rule="evenodd" d="M6 5.5h12a.5.5 0 01.5.5v7H14a2 2 0 11-4 0H5.5V6a.5.5 0 01.5-.5zm-.5 9V18a.5.5 0 00.5.5h12a.5.5 0 00.5-.5v-3.5h-3.337a3.5 3.5 0 01-6.326 0H5.5zM4 13V6a2 2 0 012-2h12a2 2 0 012 2v12a2 2 0 01-2 2H6a2 2 0 01-2-2v-5z" clip-rule="evenodd"/>
```
Comment (jasmussen) [inline packages/icons/src/library/inbox.js:8]: Do we actually need the fill-rule here?

File: packages/icons/src/library/trending-down.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/**
+ * WordPress dependencies
+ */
+import { SVG, Path } from '@wordpress/primitives';
+
+const trendingDown = (
+	<SVG xmlns="http://www.w3.org/2000/SVG" viewBox="0 0 24 24">
+	  <path d="M3.496 8.555l1.01-1.11 5.004 4.55 4.025-3.521 4.715 4.715V10h1.5v5.75H14v-1.5h3.19l-3.724-3.723-3.975 3.478-5.995-5.45z" />
```
Comment (jasmussen) [inline packages/icons/src/library/trending-down.js:8]: The tab indent looks wrong here. Are you using vs code with a linter installed?

**Review by jasmussen — COMMENTED** [review]
Nice work here. 

I left some comments on the code: all path elements need to be capitalized to work in the native apps, tabs not spaces, and there were some fill rules I think can be removed.

I'd love to more seriously explore a build tool for this, so it's easier up drop in an svg and have these js components baked automatically. But in the mean time there's a bit of manual work.

For me it helps a lot to have vs code with a lining extension installed, that will highlight most issues — not the fill rule, but the other ones. They need to be addressed for the checks to pass anyway.

I'll defer to Pablo on the visuals, otherwise seems like a nice PR!

File: packages/icons/src/library/inbox.js (hunk 2 above)
Comment (jameskoster) [reply packages/icons/src/library/inbox.js:8]: Yes, without it you get this:

fill

**Comment by pablohoneyhoney** [comment]
Apologies I missed this. Looking good @jameskoster !

Here a few details that could help:

Less notches for the receipt
Icon Grid-3

A tiny bit less inflated
Icon Grid-2

I wonder if this could be a simpler way to refer to Institution. 
Icon Grid-1

Less lines or corners
Icon Grid

And a rounded touch on the tail of the line
Icon Grid-1
Icon Grid

**Comment by jameskoster** [comment]
Thanks for taking a look @pablohoneyhoney. I updated all the icons except institute. To me your example looks more like "Home", which I think is a different icon altogether. We should probably add that one too though, in fact I will have a need of that quite soon. But I'd prefer to do it in a separate PR seeing as we're so close with this one :)

**Comment by pablohoneyhoney** [comment]
Sounds good @jameskoster 

The other element we could revise sooner than later is naming conventions, current _Product_ should likely be called _Box_, and _Review_ called _Star_, to avoid confusion or wrong use of those semantics. Worth refining overall to be more specific.

**Comment by jameskoster** [comment]
Agreed, and done :)

**Comment by mtias** [comment]
Should dollar, pound, euro be prefixed with "currency-"? Pound in particular would be ambiguous otherwise. I'd also imagine we'd grow it with other currency symbols / localizations.

File: packages/icons/src/index.js [hunk 4]
```diff
@@ -57,6 +59,8 @@ export { default as grid } from './library/grid';
 export { default as group } from './library/group';
 export { default as heading } from './library/heading';
 export { default as help } from './library/help';
+export { default as inbox } from './library/inbox';
+export { default as institute } from './library/institute';
```
Comment (mtias) [inline packages/icons/src/index.js:63]: "Institution" seems better since "institute" can be mistaken as verb, no?

**Review by mtias — APPROVED** [review]
A few notes, but this is looking good.

**Comment by jasmussen** [comment]
Took a good look at the names of these icons, which is the most important because the vectors are the easiest to change of the lot, and those look good too.

Ship it! All this needs is a rebase. Let me know if you need a hand with that!

**Comment by jameskoster** [comment]
Rebase complete! :)

**Comment by jasmussen** [comment]
Press that green button and congrats + thanks for the PR!

**Pull request merged**